### PR TITLE
[filebeat] Add SSL and AUTH username support for Redis input

### LINKED ISSUE #39412: filebeat/input/redis: Add AUTH (username) and SSL/TLS support
Redis metricbeat module already has the support of AUTH (username) and SSL/TLS for `key`, `keyspace`, and `info` metricsets but not for `slowlog` fileset.

```[tasklist]
### Tasks
- [ ] Add AUTH (username) and SSL/TLS support for Redis filebeat input to provide support for `slowlog` dataset of Redis integration.
```

Related PR: https://github.com/elastic/beats/pull/35240

### PR DESCRIPTION
**Pull request by ghost**
## Overview

This PR adds SSL and username AUTH support for `redis` input.

## Checklist

- [x] My code follows the style guidelines of this project
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Closes: #39412

### CODE REVIEW

**Comment by gpop63** [comment]
### How to test

Generate certificates `sh certs.sh`.

<details><summary><strong>openssl.cnf</strong></summary>
<p>

```
[req]
distinguished_name = req_distinguished_name
req_extensions = v3_req

[req_distinguished_name]
commonName = Common Name
commonName_default = localhost

[v3_req]
subjectAltName = @alt_names

[alt_names]
IP.1 = 0.0.0.0
DNS.1 = localhost
```

</p>
</details>

<details><summary><strong>certs.sh</strong></summary>
<p>

```sh
#!/bin/bash

# Set the certificates directory
CERTS_DIR="./certs"

# Create certs directory
mkdir -p "$CERTS_DIR"

# Set OpenSSL subject information
OPENSSL_SUBJ="/C=US/ST=California/L=Santa Clara"
OPENSSL_CA="${OPENSSL_SUBJ}/CN=fake-CA"
OPENSSL_SERVER="${OPENSSL_SUBJ}/CN=fake-server"

# Generate root CA
openssl genrsa 2048 > "$CERTS_DIR/root-ca-key.pem"

openssl req -new -x509 -nodes -days 3600 \
    -subj "${OPENSSL_CA}" \
    -key "$CERTS_DIR/root-ca-key.pem" -out "$CERTS_DIR/root-ca.pem"

# Generate server certificate
openssl req -newkey rsa:2048 -days 3600 -nodes \
    -subj "${OPENSSL_SERVER}" \
    -keyout "$CERTS_DIR/server-key.pem" -out "$CERTS_DIR/server-req.pem" \
    -config openssl.cnf

openssl rsa -in "$CERTS_DIR/server-key.pem" -out "$CERTS_DIR/server-key.pem"

openssl x509 -req -in "$CERTS_DIR/server-req.pem" -days 3600 \
    -CA "$CERTS_DIR/root-ca.pem" -CAkey "$CERTS_DIR/root-ca-key.pem" \
    -set_serial 01 -out "$CERTS_DIR/server-cert.pem" \
    -extensions v3_req -extfile openssl.cnf

openssl verify -CAfile "$CERTS_DIR/root-ca.pem" "$CERTS_DIR/server-cert.pem"

echo "Certificate generation complete. Certificates are stored in $CERTS_DIR"
```

</p>
</details>

<details><summary><strong>docker-compose.redis.yml</strong></summary>
<p>

```yaml
services:
  redis:
    image: redis:latest
    command: redis-server /usr/local/etc/redis/redis.conf
    ports:
      - "6379:6379"
    volumes:
      - ./redis.conf:/usr/local/etc/redis/redis.conf
      - ../certs:/certs
    environment:
      - REDIS_PASSWORD=password
```

</p>
</details>

<details><summary><strong>redis.conf</strong></summary>
<p>

```
port 0
tls-port 6379
tls-cert-file /certs/server-cert.pem
tls-key-file /certs/server-key.pem
tls-ca-cert-file /certs/root-ca.pem
requirepass password
user default on >password ~* &* +@all
slowlog-log-slower-than 100
slowlog-max-len 128
```

</p>
</details>

`redis.yml`

```yaml
- module: redis
  slowlog:
    enabled: true
    var.hosts: ["localhost:6379"]
    var.password: password
```

`slowlog.yml`

```yaml
type: redis
hosts:
{{ range $i, $host := .hosts }}
 - {{$host}}
{{ end }}
password: {{ .password }}
ssl.enabled: true
ssl.certificate_authorities: ["/path/to/certs/root-ca.pem"]
ssl.certificate: "/path/to/certs/server-cert.pem"
ssl.key: "/path/to/certs/server-key.pem"
processors:
  - add_fields:
      target: ''
      fields:
        ecs.version: 1.12.0
```

**Comment by shmsr** [comment]
The changes look really good. I'll take a good look again in the morning.

Appreciate making the script and using openssl directly. But if someone wants to follow easier steps (for testing) then certificate generation is included in Redis' repo. See: https://github.com/elastic/beats/pull/35240#issue-1686583037 (see: how to test locally)

File: CHANGELOG.next.asciidoc [hunk 1]
```diff
@@ -300,6 +300,7 @@ https://github.com/elastic/beats/compare/v8.8.1\...main[Check the HEAD diff]
 - Relax constraint on Base DN in entity analytics Active Directory provider. {pull}40054[40054]
 - Enhance input state reporting for CEL evaluations that return a single error object in events. {pull}40083[40083]
 - Allow absent credentials when using GCS with Application Default Credentials. {issue}39977[39977] {pull}40072[40072]
+- Add SSL support for Redis input. {pull}40111[40111]
```
Comment (shmsr) [inline CHANGELOG.next.asciidoc:303]:
```suggestion
- Add SSL and username support for Redis input, now the input includes support for Redis 6.0+. {pull}40111[40111]
```

Comment (shmsr) [reply CHANGELOG.next.asciidoc:303]: The username came in Redis 6.0.

So, effectively we now support Redis 6.0 and above as well. See, if input docs needs update?

**Review by shmsr — CHANGES_REQUESTED** [review]

**Comment by shmsr** [comment]
Please update the doc: https://github.com/elastic/beats/blob/21457f1485f83a02c78e8002fea17b385d20b139/filebeat/docs/inputs/input-redis.asciidoc

**Comment by shmsr** [comment]
> Please update the doc: https://github.com/elastic/beats/blob/21457f1485f83a02c78e8002fea17b385d20b139/filebeat/docs/inputs/input-redis.asciidoc

Add username doc as well. It is important.

**Review by shmsr — APPROVED** [review]

**Review by belimawr — COMMENTED** [review]
The implementation looks good, but I'm missing some tests. An integration tests would probably be better.

**Review by shmsr — APPROVED** [review]

**Review by rdner — APPROVED** [review]

**Pull request merged**